PurchasesCoreSwift/Networking/Backend.swift

```diff
@@ -0,0 +1,614 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  Backend.swift
+//
+//  Created by Joshua Liebowitz on 8/2/21.
+
+import Foundation
+
+public typealias SubscriberAttributeDict = [String: SubscriberAttribute]
+public typealias BackendPurchaserInfoResponseHandler = (PurchaserInfo?, Error?) -> Void
+public typealias IntroEligibilityResponseHandler = ([String: IntroEligibility]) -> Void
+public typealias OfferingsResponseHandler = ([String: Any]?, Error?) -> Void
+public typealias OfferSigningResponseHandler = (String?, String?, UUID?, NSNumber?, Error?) -> Void
+
+// swiftlint:disable type_body_length file_length
+// TODO(post-migration): Make this internal again, and all the other things too
+@objc(RCBackend) public class Backend: NSObject {
+
+    @objc public static let RCSuccessfullySyncedKey: NSError.UserInfoKey = "rc_successfullySynced"
+    @objc public static let RCAttributeErrorsKey = "attribute_errors"
+    static let RCAttributeErrorsResponseKey = "attributes_error_response"
+
+    private let httpClient: HTTPClient
+    private let apiKey: String
+
+    // callbackQueue controls access to both offeringsCallbacksCache and purchaserInfoCallbacksCache
+    private let callbackQueue = DispatchQueue(label: "Backend callbackQueue")
```

Comment:
Would it make sense to have one queue per dictionary? Actually, the code is correct but I'm thinking about a scenario where this could fail 🤔

Reply:
I think in general, yes. In this case, we already spin up a few queues for locking purposes. Since it's not necessary to have two (one won't impact performance/correctness), and one queue reduces complexity, I think keeping it to one is best.